### DISCUSSION TITLE
IRSend Protocol "CARRIER_AC84"

### DISCUSSION THREAD

**Discussion by RBroks** [arendst/Tasmota#20815] [Support request] [answered]
Hello i have been working around IR transmitter and get stuck with IRSend to CARRIER_AC84, how i understood Tasmota only suport 64bit decoding, does it mean that if the protocol is supported from IRRemote8266, but is bigger than 64bit, it is not supported in Tasmota? I find out that IR diode is no eaven blinking if the telegram is bigger than 64bit? Is it glich or it have to be like this?
And is there any walkaround?
Thank you

**Comment by s-hadinger** [comment]
Tasmota uses IRRemoteESP8266. The decoding is done in this lib, so there are no such limitations. This implies of course that you use high level parameters. There is no support for raw values bigger than 64 bits.

What commands did you try?

**Reply by RBroks** [reply]
This is what i read, and try to send
{"IrReceived":{"Protocol":"CARRIER_AC84","Bits":84,"Data":"0x0C04233200120012001204","Repeat":0}}

**Reply by sfromis** [reply]
*Exactly* what command do you use to try a send? If you use the JSON format, you should not include the `IrReceived` level.

**Reply by RBroks** [reply]
irsend {"Protocol":"CARRIER_AC84","Bits":84,"Data":0x0C04233200120012001204}

**Reply by s-hadinger** [reply]
I'm surprised it is not decoded as `irhvac` message in the first place

**Reply by RBroks** [reply]
I tryied irhvac as well it shows {"IRHVAC":"Wrong Vendor (LG|COOLIX|DAIKIN......

**Reply by RBroks** [reply]
Could someone help with some solution? If there is one..

**Reply by s-hadinger** [reply]
The support for `CARRIER_AC84` in IRRemoteESP8266 is minimalistic, as stated in doc: "Status: BETA / Untested but probably works."

Unfortunately the encoding/decoding is not implemented, and it supports only the raw data stream. In addition, Tasmota does not support data stream greater than 64 bits.

I'm afraid that there is no easy solution:
- either ask IRRemoteESP8266 to implement fully the CARRIER_AC84 protocol
- or extend the Tasmota API to accept arbitrary long input stream (I don't have time for this in the short term)

**Comment by s-hadinger** [comment]
Hold on, it may not be that difficult to go beyond 64 bits after all.

**Comment by s-hadinger** [comment, marked as answer]
Please try again with the latest version.

`irsend {"Protocol":"CARRIER_AC84","Bits":84,"Data":0x0C04233200120012001204}`

should now work.

**Reply by RBroks** [reply]
Nice, it is sending now, will try out tomorrow on AC.

**Comment by RBroks** [comment]
I did some testing today, to be corret it is fancoil model Sabiana CVP-T-3V, it comes with remote RR03 and was decoded with IRRemote8266 library as Carrier_AC84. Tried several commands to send responded to all of them, so probably it will work as it should, i cant test it fully right now because the fancoil system is under contraction and have no coolant in system. Thank you for made changes in code to support bigger code than 64bit. I am sure that will help to a lot with partly supported protocols.

**Reply by sfromis** [reply]
Well, working out what those 176 bits should be may be "less than trivial", as it is typical for HVAC  device remotes to send a payload with all values in one go, instead of being button presses like with typical simple IR remotes.

**Reply by RBroks** [reply]
I know, it is good for me i dont need all commands anyways will read the usefull ones and use them. Just give remote number un fancoil model for other to know that it is working.

**Reply by sfromis** [reply]
Well, my point is that what you send via IR is (probably) not a command but a new combined state.

**Reply by RBroks** [reply]
Yea it is like you say my english is not so good. It send full settings every time.
